snuba/clickhouse.py

```diff
@@ -374,246 +374,77 @@ def for_schema(self):
-
-    # http interface
-    ('http_method', Nullable(String())),
-    ('http_referer', Nullable(String())),
-
-    # exception interface
-    ('exception_stacks', Nested([
-        ('type', Nullable(String())),
-        ('value', Nullable(String())),
-        ('mechanism_type', Nullable(String())),
-        ('mechanism_handled', Nullable(UInt(8))),
-    ])),
-    ('exception_frames', Nested([
-        ('abs_path', Nullable(String())),
-        ('filename', Nullable(String())),
-        ('package', Nullable(String())),
-        ('module', Nullable(String())),
-        ('function', Nullable(String())),
-        ('in_app', Nullable(UInt(8))),
-        ('colno', Nullable(UInt(32))),
-        ('lineno', Nullable(UInt(32))),
-        ('stack_level', UInt(16)),
-    ])),
-
-    # These are columns we added later in the life of the (current) production
-    # database. They don't necessarily belong here in a logical/readability sense
-    # but they are here to match the order of columns in production becase
-    # `insert_distributed_sync` is very sensitive to column existence and ordering.
-    ('culprit', Nullable(String())),
-    ('sdk_integrations', Array(String())),
-    ('modules', Nested([
-        ('name', String()),
-        ('version', String()),
-    ])),
-]
-
-# The set of columns, and associated keys that have been promoted
-# to the top level table namespace.
-PROMOTED_COLS = {
-    'tags': frozenset(col.flattened for col in (PROMOTED_TAG_COLUMNS + PROMOTED_CONTEXT_TAG_COLUMNS)),
-    'contexts': frozenset(col.flattened for col in PROMOTED_CONTEXT_COLUMNS),
-}
-
-# For every applicable promoted column,  a map of translations from the column
-# name  we save in the database to the tag we receive in the query.
-COLUMN_TAG_MAP = {
-    'tags': {col.flattened: col.flattened.replace('_', '.') for col in PROMOTED_CONTEXT_TAG_COLUMNS},
-    'contexts': {},
-}
-
-# And a reverse map from the tags the client expects to the database columns
-TAG_COLUMN_MAP = {
-    col: dict(map(reversed, trans.items())) for col, trans in COLUMN_TAG_MAP.items()
-}
-
-# The canonical list of foo.bar strings that you can send as a `tags[foo.bar]` query
-# and they can/will use a promoted column.
-PROMOTED_TAGS = {
-    col: [COLUMN_TAG_MAP[col].get(x, x) for x in PROMOTED_COLS[col]]
-    for col in PROMOTED_COLS
-}
-
-
-def get_table_definition(name, engine, columns=ALL_COLUMNS):
-    return """
-    CREATE TABLE IF NOT EXISTS %(name)s (%(columns)s) ENGINE = %(engine)s""" % {
-        'columns': columns.for_schema(),
-        'engine': engine,
-        'name': name,
-    }
+# These functions are temporary and are used to help the migraiton of the DDL to
+# the dataset abstractions. Please do not introduce dependencies on them.
+# We need them right now because there is plenty of code that depends on the DDL of the
+# default dataset but that does not have access to the dataset object.
 
+def get_metadata_columns():
+    from snuba.datasets.factory import get_dataset
```

Comment:
right, unfortunately I cannot add the import at the top of the file since snuba.dataset.factory already depends on snuba.clickhouse (that is the right way the dependency should be, not the other way around).
But during the refactoring there are still dependencies between each other.
On the bright side these methods (together with their imports) are going away in the PR on top of this.